addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/discovery/KodiDiscoveryParticipant.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.kodi.internal.discovery;
+
+import static org.openhab.binding.kodi.KodiBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.UpnpDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.jupnp.model.meta.RemoteDevice;
+import org.osgi.service.component.ComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An UpnpDiscoveryParticipant which allows to discover Kodi AVRs.
+ *
+ * @author Paul Frank
+ *
+ */
+public class KodiDiscoveryParticipant implements UpnpDiscoveryParticipant {
```

Comment:
mDNS is sometimes more reliable and potentially also works on IPv6 networks. Some routers also block local UPnP traffic. It seemed that ZeroConf/mDNS announcement is active by default on Kodi, while UPnP needs to be manually activated, so this might be another reason to consider it. But I cannot give any recommendation, I just wondered, if you weighed the one against the other already. What do other Kodi clients use? Probably best to go with the mainstream.